libr/io/p/io_uf2.c

```diff
@@ -0,0 +1,454 @@
+	{ 0x3d308e94, "ESP32P4", "ESP32-P4", NULL, NULL, -1 },
+	{ 0x4b684d71, "MaixPlay-U4", "Sipeed MaixPlay-U4(BL618)", NULL, NULL, -1 },
+	{ 0x4c71240a, "STM32G4", "ST STM32G4xx", NULL, NULL, -1 },
+	{ 0x4f6ace52, "CSK4", "LISTENAI CSK300x/400x", NULL, NULL, -1 },
+	{ 0x4fb2d5bd, "MIMXRT10XX", "NXP i.MX RT10XX", NULL, NULL, -1 },
+	{ 0x53b80f00, "STM32F7", "ST STM32F7xx", NULL, NULL, -1 },
+	{ 0x540ddf62, "ESP32C6", "ESP32-C6", NULL, NULL, -1 },
+	{ 0x55114460, "SAMD51", "Microchip (Atmel) SAMD51", NULL, NULL, -1 },
+	{ 0x57755a57, "STM32F4", "ST STM32F4xx", NULL, NULL, -1 },
+	{ 0x5a18069b, "FX2", "Cypress FX2", NULL, NULL, -1 },
+	{ 0x5d1a0a2e, "STM32F2", "ST STM32F2xx", NULL, NULL, -1 },
+	{ 0x5ee21072, "STM32F1", "ST STM32F103", NULL, NULL, -1 },
+	{ 0x621e937a, "NRF52833", "Nordic NRF52833", NULL, NULL, -1 },
+	{ 0x647824b6, "STM32F0", "ST STM32F0xx", NULL, NULL, -1 },
+	{ 0x68ed2b88, "SAMD21", "Microchip (Atmel) SAMD21", NULL, NULL, -1 },
+	{ 0x699b62ec, "CH32V", "WCH CH32V2xx and CH32V3xx", NULL, NULL, -1 },
+	{ 0x6b846188, "STM32F3", "ST STM32F3xx", NULL, NULL, -1 },
+	{ 0x6d0922fa, "STM32F407", "ST STM32F407", NULL, NULL, -1 },
+	{ 0x6db66082, "STM32H7", "ST STM32H7xx", NULL, NULL, -1 },
+	{ 0x6e7348a8, "CSK6", "LISTENAI CSK60xx", NULL, NULL, -1 },
+	{ 0x6f752678, "NRF52832xxAB", "Nordic NRF52832xxAB", NULL, NULL, -1 },
+	{ 0x70d16653, "STM32WB", "ST STM32WBxx", NULL, NULL, -1 },
+	{ 0x72721d4e, "NRF52832xxAA", "Nordic NRF52832xxAA", NULL, NULL, -1 },
+	{ 0x77d850c4, "ESP32C61", "ESP32-C61", NULL, NULL, -1 },
+	{ 0x7eab61ed, "ESP8266", "ESP8266", NULL, NULL, -1 },
+	{ 0x7f83e793, "KL32L2", "NXP KL32L2x", NULL, NULL, -1 },
+	{ 0x8fb060fe, "STM32F407VG", "ST STM32F407VG", NULL, NULL, -1 },
+	{ 0x9517422f, "RZA1LU", "Renesas RZ/A1LU (R7S7210xx)", NULL, NULL, -1 },
+	{ 0x9af03e33, "GD32VF103", "GigaDevice GD32VF103", NULL, NULL, -1 },
+	{ 0xa0c97b8e, "AT32F415", "ArteryTek AT32F415", NULL, NULL, -1 },
+	{ 0xada52840, "NRF52840", "Nordic NRF52840", NULL, NULL, -1 },
+	{ 0xbfdd4eee, "ESP32S2", "ESP32-S2", "xtensa", NULL, 32 },
+	{ 0xc47e5767, "ESP32S3", "ESP32-S3", "xtensa", NULL, 32 },
+	{ 0xd42ba06c, "ESP32C3", "ESP32-C3", "riscv", NULL, 32 },
+	{ 0xe48bff56, "RP2040", "Raspberry Pi RP2040", "arm", "cortex", 16 },
+	{ 0xf71c0343, "ESP32C5", "ESP32-C5", NULL, NULL, -1 },
+};
+
+static int binary_search(UF2_Family *A, ut32 key, int imin, int imax) {
+	int imid;
+
+	// continually narrow search until just one element remains
+	while (imin < imax) {
+		R_LOG_DEBUG ("uf2: binary search %08x [%d, %d]", key, imin, imax);
+		imid = (imin + imax) / 2;
+		if (A[imid].id < key) {
+			imin = imid + 1;
+		} else {
+			imax = imid;
+		}
+	}
+
+	// At exit of while:
+	//   if A[] is empty, then imax < imin
+	//   otherwise imax == imin
+
+	// deferred test for equality
+	if ((imax == imin) && (A[imin].id == key)) {
+		return imin;
+	}
+	return -1;
+}
+
+static void process_family_id(RCore *core, ut32 family_id) {
```

Comment:
```suggestion
static void process_family_id(RIO *io, ut32 family_id) {
```

pass io, this way the function have also access to the RCoreBind struct and call io->coreb.cmdf(io->coreb.core, ...)